core/src/env/engine/on_chain/ext.rs

```diff
@@ -67,38 +156,65 @@ pub fn deposit_event(topics: &[u8], data: &[u8]) {
     }
 }
 
-pub fn set_storage(key: &[u8], value: Option<&[u8]>) {
-    match value {
-        Some(value) => unsafe {
-            sys::ext_set_storage(
-                key.as_ptr() as u32,
-                1,
-                value.as_ptr() as u32,
-                value.len() as u32,
-            )
-        },
-        None => unsafe { sys::ext_set_storage(key.as_ptr() as u32, 0, 0, 0) },
+pub fn set_storage(key: &[u8], encoded_value: &[u8]) {
+    unsafe {
+        sys::ext_set_storage(
+            key.as_ptr() as u32,
+            1,
+            encoded_value.as_ptr() as u32,
+            encoded_value.len() as u32,
+        )
     }
 }
 
-pub fn get_storage(key: &[u8]) -> RetCode {
-    unsafe { sys::ext_get_storage(key.as_ptr() as u32) }.into()
+pub fn clear_storage(key: &[u8]) {
+    unsafe { sys::ext_set_storage(key.as_ptr() as u32, 0, 0, 0) }
 }
 
-pub fn get_runtime_storage(key: &[u8]) -> RetCode {
-    unsafe { sys::ext_get_runtime_storage(key.as_ptr() as u32, key.len() as u32) }.into()
+pub fn get_storage(key: &[u8]) -> Result<()> {
```

Comment:
Missing comments for these pub fns

Reply:
I think it is okay for them to be missing doc comments since they are not really public to the outside of the on-chain module. Also they are all just thin wrappers around their corresponding `ext_*` functions which do have extensive docs.